tests/integration_tests/performance/test_snapshot_restore_performance.py

```diff
@@ -106,80 +94,77 @@ def default_lambda_consumer(env_id, workload):
-        ifaces = net_ifaces[:nets]
-
-    # Create a fresh microvm from artifacts.
-    vm_instance = vm_builder.build(
-        kernel=context.kernel,
-        disks=[rw_disk],
-        ssh_key=ssh_key,
-        config=context.microvm,
-        net_ifaces=ifaces,
-        use_ramdisk=True,
-        io_engine="Sync",
-        monitor_memory=False,
-    )
-    basevm = vm_instance.vm
-    response = basevm.machine_cfg.put(
+    scratch_drives = get_scratch_drives()
+    ifaces = net_ifaces[:nets]
+
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn(use_ramdisk=True)
+    vm.basic_config(
         vcpu_count=vcpus,
         mem_size_mib=mem_size,
+        rootfs_io_engine="Sync",
+        use_initrd=True,
     )
-    assert basevm.api_session.is_status_no_content(response.status_code)
+
+    for iface in ifaces:
+        vm.create_tap_and_ssh_config(
+            host_ip=iface.host_ip,
+            guest_ip=iface.guest_ip,
+            netmask_len=iface.netmask,
+            tapname=iface.tap_name,
+        )
+        response = vm.network.put(
+            iface_id=iface.dev_name,
+            host_dev_name=iface.tap_name,
+            guest_mac=iface.guest_mac,
+        )
+        assert vm.api_session.is_status_no_content(response.status_code)
 
     extra_disk_paths = []
     if blocks > 1:
         for name, diskfile in scratch_drives[: (blocks - 1)]:
-            basevm.add_drive(name, diskfile.path, use_ramdisk=True, io_engine="Sync")
+            vm.add_drive(name, diskfile.path, use_ramdisk=True, io_engine="Sync")
             extra_disk_paths.append(diskfile.path)
         assert len(extra_disk_paths) > 0
 
     if all_devices:
-        response = basevm.balloon.put(
+        response = vm.balloon.put(
             amount_mib=0, deflate_on_oom=True, stats_polling_interval_s=1
         )
-        assert basevm.api_session.is_status_no_content(response.status_code)
+        assert vm.api_session.is_status_no_content(response.status_code)
 
-        response = basevm.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path="/v.sock")
-        assert basevm.api_session.is_status_no_content(response.status_code)
+        response = vm.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path="/v.sock")
+        assert vm.api_session.is_status_no_content(response.status_code)
 
-    basevm.start()
+    vm.start()
 
-    logger.info(
-        'Testing with microvm: "{}", kernel {}, disk {}'.format(
-            microvm_spec, context.kernel.name(), context.disk.name()
-        )
-    )
     # Create a snapshot builder from a microvm.
-    snapshot_builder = SnapshotBuilder(basevm)
+    snapshot_builder = SnapshotBuilder(vm)
     full_snapshot = snapshot_builder.create(
-        [rw_disk.local_path()] + extra_disk_paths,
-        ssh_key,
+        [vm.rootfs_file] + extra_disk_paths,
+        rootfs.ssh_key(),
         SnapshotType.FULL,
         net_ifaces=ifaces,
-        use_ramdisk=True,
+        # use_ramdisk=True,
```

Comment:
Left over comment?

Reply:
yep, actually I just noticed it before, I need to uncomment that.